core/src/main/java/hudson/util/Iterators.java

```diff
@@ -410,4 +413,20 @@ public static void skip(@NonNull Iterator<?> iterator, int count) {
         }
     }
 
+    /**
+     *
+     * @param iterator the iterator
+     * @param position – position of the element to return
+     * @return the element at the specified position in iterator
+     */
+    public static <T> T get(Iterator<T> iterator, int position){
```

Comment:
I'm curious why we're introducing this new API. Was there a particular need for this?

Reply:
so we can remove use of  `com.google.common.collect.Iterators` from RunList